sphinx_gallery/gen_gallery.py

```diff
@@ -254,6 +255,20 @@ def call_memory(func):
     if (not isinstance(last_cell, str)) and (last_cell is not None):
         raise ConfigError("The 'last_notebook_cell' parameter must be type str"
                           " or None, found type %s" % type(last_cell))
+    # Check pypandoc
+    pypandoc = gallery_conf['pypandoc']
+    if not isinstance(pypandoc, (dict, bool)):
+        raise ConfigError("'pypandoc' must be a dict or bool, got: "
+                          "%s" % (type(pypandoc),))
+    if isinstance(pypandoc, dict):
+        accepted_keys = ('extra_args', 'filters')
+        for key in pypandoc:
+            if key not in accepted_keys:
+                raise ConfigError("'pypandoc' only accepts the following key "
+                                  "values: %s, got: %s."
+                                  % (accepted_keys, key))
+    gallery_conf['pypandoc'] = dict() if pypandoc is True else pypandoc
```

Comment:
The place you want to check-and-log/convert is here (once, at the beginning)
```suggestion
    gallery_conf['pypandoc'] = dict() if pypandoc is True else pypandoc
    if (isinstance(gallery_conf['pypandoc'], dict) and \
            not _has_pypandoc(log_result=True)):
        gallery_conf['pypandoc'] = False
```
        
```